pipeline/scripts/mac-sign.js

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+const { execSync } = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const globby = require('globby');
+
+const identityPath = process.argv[2];
+const givenEntitlementsPath = process.argv[3];
+
+function sign(pathToSign, withEntitlements) {
+    const entitlements = withEntitlements ? `--entitlements "${givenEntitlementsPath}"` : '';
+    console.log(
+        `executing: codesign -s ***** --timestamp --force --options runtime ${entitlements} ${pathToSign}`,
+    );
+    execSync(
+        `codesign -s ${identityPath} --timestamp --force --options runtime ${entitlements} "${pathToSign}"`,
+        { stdio: 'inherit' },
+    );
+}
+
+function signAsset(asset, pathToAsset) {
+    const fullAssetPath = path.join(pathToAsset, asset);
+    sign(fullAssetPath, false);
+}
+
+const appLocations = [
+    path.resolve('./drop/electron/unified-canary/packed/mac/'),
+    path.resolve('./drop/electron/unified-insider/packed/mac/'),
+    path.resolve('./drop/electron/unified-production/packed/mac/'),
+];
+
+appLocations.forEach(dir => {
+    const files = fs.readdirSync(dir);
+    const app = files.find(f => path.extname(f) === '.app');
```

Comment:
Would it be possible to include a comment explaining how you identified the specific set of stuff that needed to be signed and how you chose which to sign with/without entitlements. Did you find it documented somewhere? Did you follow another program's example? Did you find some through experimentation, and if so, what tool(s) did you use to identify missed cases?

Ideally, if a future Electron update adds some new case that needs signing, it'd be good if a future maintainer had enough information to quickly identify what they'd need to change.

Reply:
This is tough to fully capture in a comment. I did my best to summarize but I'm planning on writing separate documentation wrt some of this anyways and likely going over it in eng-discussion, which might be better-suited.